### PR TITLE
Digital root

### PR DESCRIPTION
**Pull request by meg-1**
### Describe your change:

**added the 'digital_root' method to maths/sum_of_digits.py.
a digital root is: https://en.wikipedia.org/wiki/Digital_root**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x]  This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by meg-1** [comment]
@poyea @cclauss , hi! thank you for your recommendations. i have closed, and recreated the previous pull requests!

**Comment by meg-1** [comment]
@Kush1101 hi! could you please check out this pull request?

**Comment by meg-1** [comment]
@Kush1101 hi! could you please check out this pull request?

File: maths/sum_of_digits.py [hunk 1]
```diff
@@ -55,9 +55,32 @@ def sum_of_digits_compact(n: int) -> int:
...
+    >>> digital_root(123)
+    6
+    >>> digital_root(0)
+    0
+    >>> digital_root("109820394")
+    -1
+    """
+
+    try:
+        return n % 9 or n and 9 if n >= 0 else -1
```
Comment (cclauss) [inline maths/sum_of_digits.py:76]: Should `n and 9` be `n & 9` ?  https://www.geeksforgeeks.org/difference-between-and-and-in-python

File: maths/sum_of_digits.py [hunk 2]
```diff
@@ -111,6 +136,14 @@ def benchmark() -> None:
...
         "seconds",
     )
+    print(
+        "> digital_root()",
+        "\t\tans =",
+        digital_root(medium_num),
+        "\ttime =",
+        timeit("z.digital_root(z.medium_num)", setup="import __main__ as z"),
+        "seconds",
+    )
```
Comment (cclauss) [inline maths/sum_of_digits.py:146]: Please use a more readable syntax like lines 111 and 112.

File: maths/sum_of_digits.py [hunk 3]
```diff
@@ -137,6 +170,14 @@ def benchmark() -> None:
...
         "seconds",
     )
+    print(
+        "> digital_root()",
+        "\tans =",
+        digital_root(large_num),
+        "\ttime =",
+        timeit("z.digital_root(z.large_num)", setup="import __main__ as z"),
+        "seconds",
+    )
```
Comment (cclauss) [inline maths/sum_of_digits.py:180]: Please use a more readable syntax like lines 111 and 112.

**Review by cclauss — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
Please run psf/black on the code as discussed in CONTRIBUTING.md.
* https://pypi.org/project/black

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**